server/compose/types/record_detect_duplicates.go

```diff
@@ -146,6 +156,30 @@ func (rule DeDupRule) checkCaseSensitiveDuplication(ctx context.Context, ls loca
 			valErr = &RecordValueErrorSet{}
 		)
 
+		existingVv := vv.FilterByName(c.Attribute)
+		if c.IsAllEqual() {
+			if rvv.Len() == existingVv.Len() {
+				rvvmap := recordValueFrequencyMap(c.Modifier, rvv)
+				existingVvmap := recordValueFrequencyMap(c.Modifier, existingVv)
+
+				if matchRecordValueFrequencyMap(rvvmap, existingVvmap) {
+					valErr.Push(RecordValueError{
+						Kind:    rule.IssueKind(),
+						Message: ls.T(ctx, "compose", rule.IssueMultivalueMessage()),
+						Meta: map[string]interface{}{
+							"field":         c.Attribute,
+							"dupValueField": c.Attribute,
+							"rule":          rule.String(),
+						},
+					})
+
+					return valErr
+				}
+			}
+
+			return nil
+		}
```

Comment:
Try to flatten this out a bit; if we go any more to the right we'll end up in space 😛